### ISSUE TITLE
simulation.currentStep is not updated after simulation.runForClockTime

### ISSUE THREAD

**Issue by cabb99** [openmm/openmm#2468]
While trying to setup a benchmark I noticed that the `simulation.currentStep` was updated after the `step` command, but not after the `runForClockTime`
```
>>> simulation.currentStep
0
>>> simulation.step(10)
>>> simulation.currentStep
10
>>> simulation.runForClockTime(0.01)
>>> simulation.currentStep
10
```
This behavior stops if I add a reporter with a high report frequency
```
>>> energy_reporter=simtk.openmm.app.StateDataReporter(sys.stdout, 1, step=True,time=True, potentialEnergy=True, temperature=True)
>>> simulation.reporters.append(energy_reporter)
>>> simulation.step(10)
#"Step","Time (ps)","Potential Energy (kJ/mole)","Temperature (K)"
11,3.6619999999998183,-1707.049991816885,298.35592659528095
...
20,3.6799999999998163,-1699.9937691134921,270.6361863977852
>>> simulation.runForClockTime(0.01)
21,3.681999999999816,-1699.0204152253252,270.1867850611275
...
507,4.6539999999997095,-1639.5257849943603,292.8506441928244
>>> simulation.currentStep
507
```
This behaviour is not fixed if I add a stateFile output
```
>>> simulation.reporters.pop(2)
<simtk.openmm.app.statedatareporter.StateDataReporter object at 0x7f21825c55f8>
>>> simulation.runForClockTime(0.001)
>>> simulation.currentStep
507
>>> simulation.runForClockTime(0.001,stateFile='temp.stt')
>>> simulation.currentStep
507
```

**Comment by peastman** [comment]
Thanks!  I see exactly what the problem is.  I'll get a fix in shortly.

**Comment by peastman** [comment]
The fix is in #2469.
